src/main/webapp/app/exercises/programming/assess/code-editor-tutor-assessment-container.component.ts

```diff
@@ -262,6 +271,22 @@ export class CodeEditorTutorAssessmentContainerComponent implements OnInit, OnDe
         }
     }
 
+    /**
+     * Load the feedback suggestions for the current submission from Athena.
+     */
+    private async loadFeedbackSuggestions(): Promise<void> {
+        this.feedbackSuggestions = (await firstValueFrom(this.athenaService.getFeedbackSuggestionsForProgramming(this.exerciseId, this.submission!.id!))) ?? [];
+        // Don't show feedback suggestions that have the same description and reference - probably it is coming from an earlier suggestion anyway
+        this.feedbackSuggestions = this.feedbackSuggestions.filter((suggestion) => {
+            for (const feedback of [...this.referencedFeedback, ...this.unreferencedFeedback]) {
```

Comment:
```suggestion
            for (const feedback of allFeedback) {
```
The list remains the same for all calls of the lambda in `filter()`. Can it be moved outside the `filter()`?
I assume the current spread will iterate and construct the list anew for every item in the outer `filter` loop over `this.feedbackSuggestions`?

Reply:
You are right! I changed it.